emission/analysis/modelling/trip_model/run_model.py

```diff
@@ -110,13 +112,21 @@ def predict_labels_with_n(
     :param model_config: optional configuration for model, for debugging purposes
     :return: a list of predictions
     """
-    user_id = trip['user_id']
+    start_model_load_time = time.process_time()
     model = _load_stored_trip_model(user_id, model_type, model_storage, model_config)
-    if model is None:
-        return [], -1
-    else:
-        predictions, n = model.predict(trip)
-        return predictions, n
+    print(f"{arrow.now()} Inside predict_labels_n: Model load time = {time.process_time() - start_model_load_time}")
```

Comment:
This clearly does capture the load time for the single load. Can you clarify how you captured the load time in the old implementation?

`2023-11-21T10:21:17.391582-07:00 Inside run_prediction_pipeline: Model load time = 14.285613000000001`

Since there were multiple loads, you would have needed to use an accumulator for it to be correct.

Reply:
Yes, I did accumulate the times for each trip in the original implementation.
I used global and module level variables in eamt.run_model and eacil.pipeline.
In eacil.pipeline I imported the variables that summed up the load times in eamt.run_model.

In eamt.run_model
```
model_load_time = 0
predict_time = 0
def predict_labels_with_n():
# Just mentioning code used for time-keeping
    start_model_load_time = time.process_time()
    model = _load_stored_trip_model(user_id, model_type, model_storage, model_config)
    global model_load_time
    model_load_time += time.process_time() - start_model_load_time
    # print(f"Inside predict_labels_n: Model load time = {time.process_time() - start_model_load_time}")
    # print(f"{arrow.now()} Inside predict_labels_n: Model load time = {model_load_time}")
```

In eacil.pipeline
```
import emission.analysis.modelling.trip_model.run_model as eamtr

def run_prediction_pipeline(): 
# Just mentioning code used for time-keeping
        print(f"{arrow.now()} Inside run_prediction_pipeline: Model load time = {eamtr.model_load_time}")
        print(f"{arrow.now()} Inside run_prediction_pipeline: Predictions complete for trip_list in time = {eamtr.predict_time}")
```